File: MorySnip/Classes/Layers/Layer_Rect.vb
Public Class Layer_Rect
    Inherits Layer

    Public Sub New(Pen As Pen, Brush As Brush, FirstPoint As Point)
        Me.Pen = Pen
        Me.Brush = Brush
        Me.Start(FirstPoint)
    End Sub

    Public Overrides Sub Paint(g As Graphics)
        Dim r As Rectangle = Me.Bounds

        If Me.Fill Then
            g.FillRectangle(Me.Brush, r)
        End If

        g.DrawRectangle(Me.Pen, r)
    End Sub
End Class

File: MorySnip/Classes/Layers/Layer_Rect.cs
using System.Drawing;

namespace Morysoft.MorySnip
{
    public class Layer_Rect : Layer
    {
        public Layer_Rect(Pen Pen, Brush Brush, Point FirstPoint)
        {
            this.Pen = Pen;
            this.Brush = Brush;
            this.Start(FirstPoint);
        }

        public override void Paint(Graphics g)
        {
            var r = this.Bounds;

            if (this.Fill)
            {
                g.FillRectangle(this.Brush, r);
            }

            g.DrawRectangle(this.Pen, r);
        }
    }
}
